# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/streams/ReadableStreamDefaultReader.cpp
# Commit: e95ebc865037
# Full Hash: e95ebc865037565faa8f2966d0ba45054f18480a
# Author: Peter Van der Beken <peterv@propagandism.org>
# Date: 2023-03-09 16:23:12
# Description:
#   Bug 1818630 - Crash in [@ mozilla::dom::ReadableStreamReaderGenericRelease]. r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D172057
# ==============================================================================

diff -r 4d5e64075977 -r e95ebc865037 dom/streams/ReadableStreamDefaultReader.cpp
--- a/dom/streams/ReadableStreamDefaultReader.cpp	Thu Mar 09 06:59:18 2023 +0000
+++ b/dom/streams/ReadableStreamDefaultReader.cpp	Thu Mar 09 07:56:34 2023 +0000
@@ -285,7 +285,7 @@
 
   // Step 4. If stream.[[state]] is "readable", reject reader.[[closedPromise]]
   // with a TypeError exception.
-  if (aReader->GetStream()->State() == ReadableStream::ReaderState::Readable) {
+  if (stream->State() == ReadableStream::ReaderState::Readable) {
     aReader->ClosedPromise()->MaybeRejectWithTypeError(
         "Releasing lock on readable stream");
   } else {
